### PR TITLE
Add explicit doesNotRecognizeSelector: to NSProxy

### PR DESCRIPTION
**Pull request by ms-jihua**
- Fixes a unit test failure on ARM
- Also cleaned up some header includes in NSProxy.mm

### CODE REVIEW

File: Frameworks/Foundation/NSProxy.mm [hunk 1]
```diff
@@ -202,12 +210,40 @@ - (BOOL)conformsToProtocol:(Protocol*)aProtocol {
     return class_conformsToProtocol(forwardClass, aProtocol);
 }
 
+// NOTE: long return value to allow nonfatal continuation to get a "valid" result (for non-fpret/non-stret calls)
+static long _throwUnrecognizedSelectorException(id self, Class isa, SEL sel) {
```
Comment (aballway) [inline Frameworks/Foundation/NSProxy.mm]: why is this helper needed if it's only being used in -[doesNotRecognizeSelector] ?

Comment (ms-jihua) [inline Frameworks/Foundation/NSProxy.mm]: it's more or less just lifted from NSObject.mm - I can consolidate it if you want, but I don't think it makes much of a difference either way

Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSProxy.mm]: I'd prefer we expose it in NSObject as an internal function and use it from there, but I'm won't be horribly hurt if we don't.

Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSProxy.mm]: nit: __throwUn*

**Comment by rajsesh-msft** [comment]
:shipit:

File: Frameworks/Foundation/NSObjectInternal.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
```
Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSObjectInternal.h]: #pragma once

**Pull request merged**